app/src/main/java/com/github/multimatum_team/multimatum/repository/FirebaseDeadlineRepository.kt

```diff
@@ -54,41 +52,66 @@ class FirebaseDeadlineRepository @Inject constructor() : DeadlineRepository {
         return Deadline(title, state, date)
     }
 
+    /**
+     * Fetch a single deadline given its unique ID.
+     */
+    override suspend fun fetch(id: DeadlineID): Deadline =
+        deserializeDeadline(deadlinesRef.document(id).get().await())
+
     /**
      * Fetch all deadlines from the database.
      */
-    override suspend fun fetchAll(): List<Deadline> =
+    override suspend fun fetchAll(): Map<DeadlineID, Deadline> =
         deadlinesRef
             .get()
             .await()
             .documents
-            .map { deserializeDeadline(it) }
+            .associate { it.id to deserializeDeadline(it) }
 
     /**
      * Insert new deadline in the database.
      */
-    override suspend fun put(deadline: Deadline) {
+    override suspend fun put(deadline: Deadline): DeadlineID =
         deadlinesRef
             .add(serializeDeadline(deadline))
             .await()
+            .id
```

Comment:
How Firebase create the id ? Can it create twice the same id if there is enough deadline ?

Reply:
No the generated ID is guaranteed to be unique.